fendermint/app/src/app.rs

```diff
@@ -666,26 +706,40 @@ where
             "process proposal"
         );
         let txs: Vec<_> = request.txs.into_iter().map(|tx| tx.to_vec()).collect();
+        let size_txs = txs.iter().map(|tx| tx.len()).sum::<usize>();
         let num_txs = txs.len();
 
-        let accept = self
-            .interpreter
-            .process(self.chain_env.clone(), txs)
-            .await
-            .context("failed to process proposal")?;
-
-        emit!(ProposalProcessed {
-            is_accepted: accept,
-            block_height: request.height.value(),
-            block_hash: request.hash.to_string().as_str(),
-            num_txs,
-            proposer: request.proposer_address.to_string().as_str()
+        let process_result = self.interpreter.process(self.chain_env.clone(), txs).await;
+
+        emit(BlockProposalReceived {
+            height: request.height.value(),
+            hash: HexEncodableBlockHash(request.hash.into()),
```

Comment:
Looks like `HexEncodableBlockHash` type is repeated in two crates. We could move somewhere more common (e.g. `ipc-observability`), under a `serde` module.

Reply:
Agreed, done.